lib/artist.rb

```diff
@@ -0,0 +1,46 @@
+
+
+class Artist
+  extend Concerns::Findable
+
+  attr_accessor :name
+  attr_reader :songs
+
+  @@all = []
+
+  def initialize(name)
+    @name = name
+    @songs = []
+  end
+
+  def self.all
+    @@all
+  end
+
+  def self.destroy_all
+    all.clear
+  end
+
+  def save
+    @@all << self
+  end
+
+  def self.create(name)
+    artist = new(name)
+    artist.save
+    artist
+  end
+
+  def add_song(song)
+    song.artist = self if song.artist.nil?
+    songs << song unless songs.include?(song)
+  end
+
+  def genres
+    songs.map do |artist|
+      artist.genre
+    end
+    .uniq
```

Comment:
It's a little weird to have a dangling line that starts with `.`. You 1. Move the `.uniq` to the end: `end.uniq`. or 2. You could save the non-unique result in a variable then call unique on it afterwards.